### DISCUSSION TITLE
How to force end the solving process.

### DISCUSSION THREAD

**Discussion by Flians** [Z3Prover/z3#6804] [Q&A]
My code is as follows:
```python
@func_set_timeout(100000)  # set the maximum runtime
def z3_sol():
  ...
  solver.check()
  ...

def main():
  try:
    z3_sol()
  except func_timeout.exceptions.FunctionTimedOut:
    # These two methods do not work. How can I end the solving process here?
    z3.Z3_finalize_memory() / z3.Z3_reset_memory() 
    print("Timeout!\n")
``` 

Could you please tell me how can I stop the solving process? Thank you.

**Comment by Joshua27** [comment]
You can set the `timeout` parameter of Z3.
Here is a list of Z3's parameters: https://microsoft.github.io/z3guide/programming/Parameters/
(You can also interrupt a Z3 context, but I think that's not what you want in this case: https://z3prover.github.io/api/html/classz3_1_1context.html#aebc9afd317d656680550a13818e71917)
